### DISCUSSION TITLE
The `url-loader` doesn't seem to work, doesn't `rspack` recommend it?

### DISCUSSION THREAD

**Discussion by peterroe** [web-infra-dev/rspack#6056] [Q&A]
Thanks this nice building tool ❤️

I met some problem with `rsbuild`. My config:

```ts
rspack: {
	module: {
		rules: [
		{
			// type: image
			test: /\.(png|jpe?g|gif|svg)$/i,
			use: [
				{
					loader: 'url-loader',
					options: {
						generator: (content, mimetype, encoding, resourcePath) => {
							// Not work
							console.log({content, mimetype, encoding, resourcePath})
							// some process
							return /* custom cdn URL */;
						}
					}
				}
			],            
		]
	}
}
```

The image is converted to `base64` without `generator` processing. It's not what I expected.

**Comment by xc2** [comment]
there's no need to use `url-loader`.

if you just want a HTTP url instead of data uri, please just try reducing [output.dataUriLimit](https://rsbuild.dev/config/output/data-uri-limit).

**Comment by peterroe** [comment]
Thank you for your response. Actually, my goal is to upload images and other resources to a self-hosted CDN service and then use the returned CDN links. 

Previously. I achieved this by configuring the `url-loader` in `webpack`. I'm not sure if there is a better way to achieve this goal In `rspack`.

 Currently, the only solution I can think of is implementing it within a webpack plugin. But it's a little more complicated.

**Reply by xc2** [reply]
rsbuild has several builtin rules for assets, thus before you add custom rules for png, you should remove the conflict builtin rules with [`tools.bundlerChain`](https://rsbuild.dev/config/tools/bundler-chain).

for example

```ts
  tools: {
    bundlerChain(chain) {
      chain.module.rules.delete("image");
    },
    rspack: {
      // your rspack configurations
    }
  },
```

alternatively, pls consider build with rspack directly

**Comment by hardfist** [comment]
url-loader is supported，can you try it in rspack directly
